doc/getting-started.md

```diff
@@ -0,0 +1,175 @@
+# Getting Started
+
+This section will serve as a simple Mio tutorial. It will start from
+generating a new [Rust](http://www.rust-lang.org/) project using
+[Cargo](https://crates.io) up to writing a simple TCP echo server and
+client.
+
+Of course, you will need Rust installed. If you haven't already, get it
+here: [rust-lang.org](https://www.rust-lang.org).
+
+> **Note:** As of the time of writing, Mio does not support windows.
+Windows support is currently in progress.
+
+## Setting up the project
+
+The first step is getting a new Cargo project setuup. In a new
+directory, run the following:
+
+```not_rust
+cargo new pingpong --bin
+cd pingpong
+```
+
+Now, open the directory in your favorite text editor. You should see the
+following files:
+
+* src/main.rs
+* Cargo.toml
+
+If you are not already familiar with Cargo, you can learn more about it
+[here](http://doc.crates.io/).
+
+Open `Cargo.toml` and add a dependency on Mio by putting the following a
+the bottom of the file:
+
+```not_rust
+[dependencies.mio]
+git = "https://github.com/carllerche/mio"
+```
+
+> Since the guide is currently written assuming some APIs that are not
+> available on crates.io yet, we depend on the master branch of Mio on
+> GitHub.
```

Comment:
Rustdoc supports a footnote style annotation, not sure if it would be better or worse to make this kind of aside into one